drivers/xen/hypervisor.c

```diff
@@ -81,11 +58,38 @@ static const char *param_name(int op)
 	return names[op];
 }
 
+/**
+ * hvm_get_parameter_maintain_dcache - function to obtain a HVM
+ * parameter value.
+ * @idx: HVM parameter index
+ * @value: Value to fill in
+ *
+ * According to Xen on ARM ABI (xen/include/public/arch-arm.h):
+ * all memory which is shared with other entities in the system
+ * (including the hypervisor and other guests) must reside in memory
+ * which is mapped as Normal Inner Write-Back Outer Write-Back
+ * Inner-Shareable.
+ *
+ * Thus, page attributes must be equally set for all the entities
+ * working with that page.
+ *
+ * Before MMU setup the data cache is turned off, so it means that
+ * manual data cache maintaining is required, because of the
+ * difference of page attributes.
+ *
+ * There is no guarantee that cache is going to be clean when
```

Comment:
not sure we need to go in this level of detail. The above description already explains why we need the cache to be maintained, so it should be clear why and when we invalidate